src/Microsoft.Fx.Portability/ObjectModel/NuGetPackageInfo.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Runtime.Versioning;
+
+namespace Microsoft.Fx.Portability.ObjectModel
+{
+    public class NuGetPackageInfo
+    {
+        private bool _hashComputed;
+        private int _hashCode;
+
+        public string AssemblyInfo { get; private set; }
+        public FrameworkName Target { get; private set; }
+        public ImmutableList<NuGetPackageId> SupportedPackages { get; private set; }
+
+        public NuGetPackageInfo(string assemblyInfo, FrameworkName target, IEnumerable<NuGetPackageId> supportedPackages)
+        {
+            AssemblyInfo = assemblyInfo;
+            Target = target;
+            var packages = supportedPackages != null ? supportedPackages : Enumerable.Empty<NuGetPackageId>();
+            SupportedPackages = packages.ToImmutableList();
+        }
+
+        public override bool Equals(object obj)
+        {
+            NuGetPackageInfo other = obj as NuGetPackageInfo;
```

Comment:
It would be clearer to either use pattern matching (`if obj is NuGetPackageInfo`) or to use `var`

Reply:
Changed